komodo/check_up_to_date_pypi.py

```diff
@@ -130,32 +133,81 @@ def insert_upgrade_proposals(upgrade_proposals, repository, releases):
         releases[package] = suggested_version
 
 
-def main():
+def run_check_up_to_date(
+    release_file,
+    repository_file,
+    python_version=(
+        f"{sys.version_info.major}."
+        f"{sys.version_info.minor}."
+        f"{sys.version_info.micro}"
+    ),
+    propose_upgrade=False,
+):
+    yaml = yaml_parser()
+    releases = load_from_file(yaml, release_file)
+    repository = load_from_file(yaml, repository_file)
+    upgrade_proposals_from_pypi = get_upgrade_proposals_from_pypi(
+        releases,
+        repository,
+        python_version,
+    )
+    if upgrade_proposals_from_pypi:
+        if propose_upgrade:
+            insert_upgrade_proposals(upgrade_proposals_from_pypi, repository, releases)
+            print(
+                "Writing upgrade proposals from pypi, "
+                "assuming nothing has changed with dependencies..."
+            )
+            with open(propose_upgrade, "w") as fout:
```

Comment:
We should always specify encoding when using  `open()`. This line normally gives an error on this, but we have added `unspecified-encoding` to `pyproject.toml` because we haven't gotten this far yet with solving pylint issues in komodo.